lib/spells/effects/Damage.cpp

```diff
@@ -115,7 +114,7 @@ void Damage::serializeJsonUnitEffect(JsonSerializeFormat & handler)
 
 int64_t Damage::damageForTarget(size_t targetIndex, const Mechanics * m, const battle::Unit * target) const
 {
-	int64_t baseDamage;
+	int64_t baseDamage = 0;
```

Comment:
not sure why this is needed - the variable is initialised in all code paths

Reply:
It is just default init of clang-tidy. I do not think than it is bad. If clang-tidy wants to add it - why not?

Reply:
IMO this is bad change.
Old behavior: if damage is not set on some code paths, compiler will issue "uninitialized" warning.
New behavior: if damage is not set on some code path, it will be zero. Not desired behavior.

Reply:
@IvanSavenko okay, will try to do modernization without cppcoreguidelines-init-variables filter.

Reply:
@IvanSavenko removed